daemon/src/db.rs

```diff
@@ -348,6 +365,444 @@ impl Connection {
 
         Ok(ids)
     }
+
+    pub async fn insert_closed_cfd(&self, cfd: close_cfds::Cfd) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        let close_cfds::Cfd {
+            id,
+            position,
+            initial_price,
+            taker_leverage,
+            n_contracts,
+            counterparty_network_identity,
+            role,
+            opening_fee,
+            initial_funding_rate,
+            initial_tx_fee_rate,
+            fee_account,
+            lock,
+            commit,
+            collaborative_settlement,
+            non_collaborative_settlement,
+            refund,
+            close_status,
+            ..
+        } = cfd;
+
+        let (lock_txid, lock_dlc_vout) = match lock {
+            Some(Lock { txid, dlc_vout }) => (txid, dlc_vout),
+            None => bail!("Closed CFD without lock"),
+        };
```

Comment:
I think we should design the API of `close_cfds::Cfd` such that this is impossible. Private field and getter perhaps?

Reply:
We should not pass a `Cfd` into here that is not actually closed.

Reply:
Yeah, I agree. I wasn't happy with all these `bail`s.